From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 28 Apr 2021 03:57:39 +0200
Subject: [klibc] malloc: Set errno on failure
Origin: https://git.kernel.org/pub/scm/libs/klibc/klibc.git/commit/?id=7f6626d12daa2f1efd9953d1f4ba2065348dc5cd

malloc() is specified to set errno = ENOMEM on failure, so do that.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 usr/klibc/malloc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/klibc/malloc.c b/usr/klibc/malloc.c
index 413b7337..bb57c9f6 100644
--- a/usr/klibc/malloc.c
+++ b/usr/klibc/malloc.c
@@ -8,6 +8,7 @@
 #include <unistd.h>
 #include <sys/mman.h>
 #include <assert.h>
+#include <errno.h>
 #include "malloc.h"
 
 /* Both the arena list and the free memory list are double linked
@@ -169,6 +170,7 @@ void *malloc(size_t size)
 #endif
 
 	if (fp == (struct free_arena_header *)MAP_FAILED) {
+		errno = ENOMEM;
 		return NULL;	/* Failed to get a block */
 	}
 
